### ISSUE TITLE
Cargo subcommand instead of shell scripts

### ISSUE THREAD

**Issue by l4l** [bjorn3/rustc_codegen_cranelift#971]
TLDR: it would be nice to have something like `cargo cl-codegen build` and similar.

For now i think it is quite clumsy to specify magic env-var(s) and running magic shell scripts by their full paths. Though, these shell scripts might be quite handy for development, they are not really convenient for usage (and probably for maintaining).

Perhaps it would be much easier to use cranelift codegen via single binary, that can be configured once with persistent file. I am not really sure about all of nitty-gritty details, but probably it can be handled with cargo sub-command with additional config file (e.g. for target, librustc_codegen_cranelift.so path, path for sysroot, etc). Do I miss something important and what do you think about that?

**Comment by CohenArthur** [comment]
I'd love to work on this, if it is available

**Comment by bjorn3** [comment]
Go ahead. Since https://github.com/bjorn3/rustc_codegen_cranelift/pull/1089 the shell scripts have been simplified a bit, so implementing this takes less work.

**Comment by bjorn3** [comment]
As of https://github.com/bjorn3/rustc_codegen_cranelift/pull/1104 the `build` directory now contains everything cg_clif needs to run. There still isn't a cargo subcommand yet though, but you can copy the `build` dir somewhere and add `cargo.sh` to your path. (optionally renaming it to something else)

**Comment by bjorn3** [comment]
This should now be a matter of renaming the produced cargo executable to cargo-clif. If you then add the build dir to your path you will be able to do `cargo clif build`. Going to rename it tomorrow.
